docs/coding-standards/js.md

```diff
@@ -1,61 +1,164 @@
-# JavaScript
+# JavaScript style guide
+## Files
+JavaScript files have the same name as the component's folder name. Test files have a `.test` suffix placed before the file extension.
 
-# Linting
+```
+checkboxes
+├── checkboxes.js
+└── checkboxes.test.js
+```
+## Skeleton
 
-GOV.UK Frontend uses [standardjs](http://standardjs.com/), an opinionated JavaScript linter. All JavaScript files follow its conventions, and it runs on CI to ensure that new pull requests are in line with them.
+```js
+import { nodeListForEach } from '../globals/common'
 
-## Running standard manually
+function Checkboxes ($module) {
+  ...
+}
 
-To check the whole codebase, run:
+Checkboxes.prototype.init = function () {
+  ...
+}
 
-```bash
-npm run lint
+export default Checkboxes
+```
+## Comments
+
+Use `/** ... */` for multi-line comments. Include a description, and specify types and values for all parameters and return values.
```

Comment:
is this [jsdoc](http://usejsdoc.org/about-getting-started.html)? If so should we mention it and have a link?

Reply:
These comment types are a de facto standard based on the javascript community best practices and the comment styles from other languages. The fact that jsdoc uses them in order to generate the documentation doesn't mean they have ownership over the standard, so I wouldn't link them.

Reply:
ok cool, I thought maybe they had defined them, my mistake. Does this style have a definition somewhere?

Reply:
not that I'm aware of, but I'll check